<?xml version="1.0" encoding="utf-8"?>	

<launch>
  <arg name="robot_prefix" default="turtlebot1"/>
  
  <node pkg="move_base" type="move_base" respawn="false" name="move_base_$(arg robot_prefix)" output="screen">
    <rosparam file="$(find turtlebot_navigation)/param/turtlebot_costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find turtlebot_navigation)/param/turtlebot_costmap_common_params.yaml" command="load" ns="local_costmap" />   
    <rosparam file="$(find turtlebot_navigation)/param/turtlebot_local_costmap_params.yaml" command="load" />   
    <rosparam file="$(find turtlebot_navigation)/param/turtlebot_global_costmap_params.yaml" command="load" />
    <rosparam file="$(find turtlebot_navigation)/param/turtlebot_dwa_local_planner_params.yaml" command="load" />
    <!-- <rosparam file="$(find turtlebot_navigation)/param/turtlebot_move_base_params.yaml" command="load" />
    <rosparam file="$(find turtlebot_navigation)/param/turtlebot_global_planner_params.yaml" command="load" />
    <rosparam file="$(find turtlebot_navigation)/param/turtlebot_navfn_global_planner_params.yaml" command="load" />
	-->
    <remap from="/map" to="$(arg robot_prefix)/gmapping"/>
  </node>
  
    <node pkg="turtlebot_navigation" type="twist_to_differential_drive" respawn="false" name="twist_to_differential_drive_$(arg robot_prefix)" output="screen">
		<param name = "robot_width" value = "0.354"/>
		<param name = "max_wheel_vel" value = "0.65"/>
		<param name = "cmd_vel_topic_name" value = "/cmd_vel"/>
		<param name = "diff_drive_vel_topic_name" value = "/$(arg robot_prefix)/wheels_vel_cmd"/>		
  </node>
  
  
</launch>
